lib/dialects/oracle/index.js

```diff
@@ -74,8 +74,8 @@ assign(Client_Oracle.prototype, {
     var client = this;
     return new Promise(function (resolver, rejecter) {
       client.driver.connect(client.connectionSettings, function (err, connection) {
-        if (err) return rejecter(err);
         Promise.promisifyAll(connection);
+        if (err) return rejecter(err);
         if (client.connectionSettings.prefetchRowCount) {
```

Comment:
https://github.com/tgriesser/knex/pull/896/files

this change wasn't made in src.